tooling/debugger/tests/debug.rs

```diff
@@ -49,4 +49,115 @@ mod tests {
         // Exit the bash session.
         dbg_session.send_line("exit").expect("Failed to quit bash session");
     }
+
+    #[test]
+    fn debugger_expected_call_stack() {
+        let nargo_bin =
+            cargo_bin("nargo").into_os_string().into_string().expect("Cannot parse nargo path");
+
+        let timeout_seconds = 30;
+        let mut dbg_session =
+            spawn_bash(Some(timeout_seconds * 1000)).expect("Could not start bash session");
+
+        let test_program_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("../../test_programs/execution_success/regression_7195");
+        let test_program_dir = test_program_path.display();
+
+        // Start debugger and test that it loads for the given program.
+        dbg_session
+            .execute(
+                &format!(
+                    "{nargo_bin} debug --raw-source-printing true --program-dir {test_program_dir} --force-brillig --expression-width 3"
+                ),
+                ".*\\Starting debugger.*",
+            )
+            .expect("Could not start debugger");
+
+        let num_steps = 16;
+        for _ in 1..=num_steps {
+            // While running the debugger, issue a "next" cmd,
+            // which should run to the program to the next source line given
+            // we haven't set any breakpoints.
+            // ">" is the debugger's prompt, so finding one
+            // after running "next" indicates that the
+            // debugger has not panicked for this step.
+            dbg_session
+                .send_line("next")
+                .expect("Debugger panicked while attempting to step through program.");
+            dbg_session
+                .exp_string(">")
+                .expect("Failed while waiting for debugger to step through program.");
+        }
+
+        let mut lines = vec![];
+        while let Ok(line) = dbg_session.read_line() {
+            if !(line.starts_with(">next") || line.starts_with("At ") || line.starts_with("...")) {
+                lines.push(line);
+            }
+        }
+
+        let lines_expected_to_contain: Vec<&str> = vec![
+            "> next",
+            "    let x = unsafe { baz(x) };",
+            "unconstrained fn baz(x: Field) -> Field {",
+            "> next",
+            "    let x = unsafe { baz(x) };",
+            "unconstrained fn baz(x: Field) -> Field {",
```

Comment:
I think that this ordering of the stacktrace is pretty confusing and backwards from as most word expect. Rather than starting at the most specific location and then showing the callstack reaching this point below it (as is done in the majority of programming languages afaik) we're always starting at the furthest away location.

Reply:
I think this might be the difference between the order from printing unwinding a stack after a panic and the order from stepping through execution with a debugger:
- Unwinding: `assert(..)\n..[inside foo]\n..foo()\n..main()`
- Line-by-line debugger: `main()\n..foo()\n..[inside foo]\n..assert(..)`